tools/cli/NearObjectCli.cxx

```diff
@@ -233,14 +238,200 @@ AddEnumOption(CLI::App* app, std::optional<EnumType>& assignTo, bool isMandatory
+
+        const std::bitset<resultReportConfigurationSize> resultReportConfigurationBits(cliData->resultReportConfigurationString);
+        for (auto i = 0; i < resultReportConfigurationSize; i++) {
+            if (resultReportConfigurationBits.test(i)) {
+                parametersData.resultReportConfig.value().insert(magic_enum::enum_value<ResultReportConfiguration>(i));
+            }
+        }
+    }
+
+    // InBandTerminationAttemptCount
+    if (parametersData.inBandTerminationAttemptCount.has_value() && parametersData.inBandTerminationAttemptCount.value() > 10) {
+        std::cerr << "Invalid InBandTerminationAttemptCount. Out of range." << std::endl;
+    }
+
+    // SubSessionId
+    if (parametersData.stsConfiguration.has_value() && parametersData.stsConfiguration.value() == StsConfiguration::DynamicWithResponderSubSessionKey) {
+        if (!parametersData.subSessionId.has_value()) {
+            std::cerr << "SubSessionId is required for Dynamic STS with Responder Sub-Session Key" << std::endl;
+        }
+    }
+
+    // BprfPhrDataRate
+    if (parametersData.prfMode.has_value() && parametersData.prfMode.value() != PrfMode::Bprf) {
+        if (parametersData.bprfPhrDataRate.has_value()) {
+            std::cerr << "Invalid BprfPhrDataRate. Value expected only in BPRF mode" << std::endl;
+        }
+    } else {
+        if (parametersData.bprfPhrDataRate.has_value() && !magic_enum::enum_contains<BprfPhrDataRate>(parametersData.bprfPhrDataRate.value())) {
+            std::cerr << "Invalid BprfPhrDataRate" << std::endl;
+        }
+    }
+}
+
+/**
+ * @brief Validates the enum application configuration parameter inputs from nocli.
+ *
+ * @tparam EnumType The type of the enumeration.
+ * @param applicationConfigurationParameter The application configuration parameter of type EnumType.
+ */
+template <typename EnumType>
+// clang-format off
+requires std::is_enum_v<EnumType>
+void
+ValidateEnumParameterValue(const EnumType& applicationConfigurationParameter)
+// clang-format on
+{
+    auto parameterType = magic_enum::enum_type_name<EnumType>();
+    if (!magic_enum::enum_contains<EnumType>(applicationConfigurationParameter)) {
+        if (parameterType == "StsPacketConfiguration") { // Special case
+            std::cerr << "Invalid RFrameConfiguration" << std::endl;
+        } else {
+            std::cerr << "Invalid " << parameterType << std::endl;
+        }
+    }
+}
+
+/**
+ * @brief Validates, displays, and returns application configuration parameters.
+ *
+ * @param cliData The nocli input data containing the application configuration parameters.
  * @return std::vector<uwb::protocol::fira::UwbApplicationConfigurationParameter>.
  */
 std::vector<uwb::protocol::fira::UwbApplicationConfigurationParameter>
-ProcessApplicationConfigurationParameters(UwbApplicationConfigurationParameterData applicationConfigurationParameterData)
+ProcessApplicationConfigurationParameters(std::shared_ptr<NearObjectCliData> cliData)
```

Comment:
nit: This being a helper method and a completely synchronous call, you could safely pass a raw reference instead. Also, could a `const` ref be passed or does this function mutate `cliData`?

```suggestion
ProcessApplicationConfigurationParameters(const NearObjectCliData& cliData)
```

Reply:
Doesn't mutate `cliData`, so I will change this to pass a raw const ref